std/src/main/scala/algebra/std/array.scala

```diff
@@ -0,0 +1,42 @@
+package algebra
+package std
+
+import scala.{specialized ⇒ sp}
+
+package object array extends ArrayInstances
+
+trait ArrayInstances {
+  implicit def arrayEq[@sp A: Eq]: Eq[Array[A]] = new ArrayEq[A]
```

Comment:
I think arrayEq should be specialized on everything. Equality comparison of e.g. byte arrays should not have boxing overhead.

For PartialOrder, just `@sp(Int, Float, Long, Double)` seems OK.